config.ts

```diff
@@ -32,66 +27,52 @@ export interface DisplayedField {
   precision?: number;
 }
 
-interface ContractConfig {
-  JUNIOR_OPERATOR: 'ALLOWANCE_OPERATOR';
-  SENIOR_OPERATOR: 'PROPORTIONAL_OPERATOR';
+interface Config {
+  rpcUrl: string;
+  etherscanUrl: string;
+  gasLimit: number;
+  transactionTimeout: number;
+  tinlakeDataBackendUrl: string;
+  isDemo: boolean;
+  network: 'Mainnet' | 'Kovan';
+  pools: Pool[];
 }
 
 const contractAddressesSchema = yup.object().required().shape({
-  DEPLOYMENT_NAME: yup.string().required(),
   ROOT_CONTRACT: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/).required(),
-  TINLAKE_CURRENCY: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/).required(),
   ACTIONS: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/).required(),
   PROXY_REGISTRY: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/).required(),
-  COLLATERAL_NFT: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/).required()
-});
-
-const nftDataDefinitionSchema = yup.object().required().shape({
-  contractCall: yup.object().required().shape({
-    outputs: yup.array(yup.object().shape({
-      name: yup.string().required().min(1),
-      type: yup.mixed<'address' | 'uint256' | 'string'>().required().oneOf(['address', 'uint256', 'string'])
-    }))
-  }),
-  displayedFields: yup.array(yup.object().shape({
-    key: yup.string().required().min(1),
-    label: yup.string().required().min(1),
-    type: yup.string().required().min(1),
-    decimals: yup.number().min(0),
-    precision: yup.number().min(0)
-  }))
+  COLLATERAL_NFT: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/),
+  DEPLOYMENT_NAME: yup.string().required()
 });
 
 const contractConfigSchema = yup.object().required().shape({
   JUNIOR_OPERATOR: yup.mixed<'ALLOWANCE_OPERATOR'>().required().oneOf(['ALLOWANCE_OPERATOR']),
   SENIOR_OPERATOR: yup.mixed<'PROPORTIONAL_OPERATOR'>().required().oneOf(['PROPORTIONAL_OPERATOR'])
 });
 
-interface Config {
-  rpcUrl: string;
-  etherscanUrl: string;
-  gasLimit: number;
-  contractAddresses: ContractAddresses;
-  nftDataDefinition: NFTDataDefinition;
-  transactionTimeout: number;
-  tinlakeDataBackendUrl: string;
-  isDemo: boolean;
-  network: 'Mainnet' | 'Kovan';
-  contractConfig:  ContractConfig;
-}
+const poolSchema = yup.object().required().shape({
+  addresses: contractAddressesSchema,
+  graph: yup.string().required(),
+  contractConfig: contractConfigSchema,
+  name: yup.string().required(),
+  description: yup.string().required(),
+  slug: yup.string().required(),
+  asset: yup.string().required()
```

Comment:
Can we call this `assetType` and supply a enum, like the following?

```suggestion
  asset: yup.mixed<'invoice' | 'bla'>().required().oneOf(['invoice', 'bla'])
```

Reply:
yes, totally.
Let's do that when we receive the spec next week